src/Microsoft.ML.StandardLearners/FactorizationMachine/FactorizationMachineTrainer.cs

```diff
@@ -323,7 +321,12 @@ private static float CalculateLossSlope(float label, float modelResponse)
                 var validFeatureColumns = data.Schema.GetColumns(RoleMappedSchema.ColumnRole.Feature);
                 Host.Assert(fieldCount == validFeatureColumns.Count);
                 for (int f = 0; f < fieldCount; f++)
-                    Host.Assert(featureColumns[f] == validFeatureColumns[f]);
+                {
+                    var featCol = featureColumns[f];
+                    var validFeatCol = validFeatureColumns[f];
+                    Host.Assert(featCol.Name == validFeatCol.Name);
+                    Host.Assert(featCol.Type == validFeatCol.Type);
```

Comment:
>== [](start = 45, length = 2)

Is it reference-equal or `Equals` ?

Reply:
We don't have such a thing as reference equals on `Schema.Column`, so the closest I could do is equals on the fields (which in the case of type is a reference equals).